pysatModels/utils/extract.py

```diff
@@ -581,49 +624,59 @@ def instrument_view_irregular_model(inst, model, inst_name, mod_name,
-    max_pts_alt = np.nanmax(points[:, update_dim])
+    # Pull out model data to be interpolated. Needed so same downselection
+    # procedure may be applied to data as to dimensions.
+    model_vars = []
+    for iname in sel_name:
+        model_vars.append(np.ravel(model[iname].values))
 
-    # Get downselection values using the minimum value
-    if min_inst_alt < max_pts_alt:
-        min_sel_val = inst[inst_var_label].min() - inst_var_delta
-    else:
-        min_sel_val = max_pts_alt - inst_var_delta
+    # Downselect all model points to those within some range of the Instrument
+    # values. For each of the model dimensions, each with a corresponding
+    # instrument variable, determine the selection criteria and store the
+    # limits.
+    for i, (iname, idelta) in enumerate(zip(inst_name, mod_var_delta)):
+        # Range from Instrument values
+        min_val = inst[iname].min()
+        max_val = inst[iname].max()
 
-    # Now get the maximum value
-    if max_inst_alt < max_pts_alt:
-        inst[inst_var_label].max() + inst_var_delta
-    else:
-        max_sel_val = max_pts_alt
+        # Range from model values
+        min_mval = np.nanmin(points[:, i + 1])
+        max_mval = np.nanmax(points[:, i + 1])
+
+        # Net range
+        min_val = np.max([min_val - idelta, min_mval])
+        max_val = np.min([max_val + idelta, max_mval])
+
+        # Determine which points are within the specified tolerance range.
+        idx, = np.where((points[:, i + 1] >= min_val)
+                        & (points[:, i + 1] <= max_val))
+
+        # Downselect model dimension values
+        points = points[idx, :]
+
+        # Perform downselection of model variables as well
+        for j, ivar in enumerate(model_vars):
+            model_vars[j] = ivar[idx]
 
-    # Perform the downselection
-    idx, = np.where((points[:, update_dim] >= min_sel_val)
-                    & (points[:, update_dim] <= max_sel_val))
-    points = points[idx, :]
     ps_mod.logger.debug('Remaining points after downselection {:d}'.format(
         len(idx)))
 
     # Create input array using inst time/position
     coords = [inst[coord] for coord in inst_name]
-    coords.insert(0, inst.index.values.astype(int))
+    coords.insert(0, inst.index.values.astype(np.int64))
     sat_pts = [inp for inp in zip(*coords)]
 
     # Perform interpolation of user desired variables
     output_names = []
-    for var in sel_name:
+    for var, idata in zip(sel_name, model_vars):
         ps_mod.logger.debug('Creating interpolation object for ' + var)
         output_names.append('_'.join((model_label, var)))
         inst[output_names[-1]] = \
-            interpolate.griddata(points,
-                                 np.ravel(model[var].values)[idx],
-                                 sat_pts,
-                                 rescale=True)
+            interpolate.griddata(points, idata, sat_pts, rescale=True)
         ps_mod.logger.debug('Complete.')
```

Comment:
Recommend adding a bit more data to the debugger statement (e.g., what has completed for this iteration).